test-framework/sudo-compliance-tests/src/lecture.rs

```diff
@@ -0,0 +1,108 @@
+use sudo_test::{Command, Env, User};
+use crate::{Result, SUDOERS_ROOT_ALL, USERNAME, SUDOERS_USER_ALL_ALL, SUDOERS_ALWAYS_LECTURE, SUDOERS_NO_LECTURE, PASSWORD};
+
+#[test]
+fn default_lecture_shown_once() -> Result<()> {
+    let expected_lecture = format!(
+        "\nWe trust you have received the usual lecture from the local System\nAdministrator. It usually boils down to these three things:\n\n    #1) Respect the privacy of others.\n    #2) Think before you type.\n    #3) With great power comes great responsibility."
+    );
+    let expected_echo = format!(
+        "Yeah!"
+    );
+
+    let env = Env(["SUDOERS_ROOT_ALL", SUDOERS_USER_ALL_ALL])
+        .user(User(USERNAME).password(PASSWORD))
+        .build()?;
+
+    let output = Command::new("sudo")
+    .args(["-S", "true"])
+    .as_user(USERNAME)
+    .stdin(PASSWORD)
+    .exec(&env)?;
+    assert_eq!(true, output.status().success());
+
+    if sudo_test::is_original_sudo() {
+        assert_contains!(
+            output.stderr(),
+            expected_lecture
+        );
+    }
+
+    let second_sudo = Command::new("sudo")
+    .as_user(USERNAME)
+    .stdin(PASSWORD)
+    .args(["-S", "echo", "Yeah!"])
+    .exec(&env)?;
+
+    assert_eq!(true, second_sudo.status().success());
+    assert_eq!(Some(0), second_sudo.status().code());
```

Comment:
it's not necessary to assert_eq `status().code()` if you already asserted `status().success()`